src/notifications/NotificationsHandler.tsx

```diff
@@ -0,0 +1,341 @@
+import { useContacts, usePrevious, useWallets } from '@/hooks';
+import { setupAndroidChannels } from '@/notifications/setupAndroidChannels';
+import messaging, {
+  FirebaseMessagingTypes,
+} from '@react-native-firebase/messaging';
+import {
+  FixedRemoteMessage,
+  MinimalNotification,
+  NotificationTypes,
+  TransactionNotificationData,
+} from '@/notifications/types';
+import { handleShowingForegroundNotification } from '@/notifications/foregroundHandler';
+import {
+  registerTokenRefreshListener,
+  saveFCMToken,
+} from '@/notifications/tokens';
+import { WALLETCONNECT_SYNC_DELAY } from '@/notifications/constants';
+import { useDispatch } from 'react-redux';
+import { requestsForTopic } from '@/redux/requests';
+import { ThunkDispatch } from 'redux-thunk';
+import store, { AppState } from '@/redux/store';
+import { AnyAction } from 'redux';
+import { NotificationStorage } from '@/notifications/deferredNotificationStorage';
+import { Navigation } from '@/navigation';
+import Routes from '@rainbow-me/routes';
+import {
+  AppState as ApplicationState,
+  AppStateStatus,
+  NativeEventSubscription,
+} from 'react-native';
+import notifee, {
+  Event as NotifeeEvent,
+  EventType,
+} from '@notifee/react-native';
+import { getProviderForNetwork } from '@/handlers/web3';
+import { ethereumUtils, isLowerCaseMatch } from '@/utils';
+import { NewTransactionOrAddCashTransaction } from '@/entities/transactions/transaction';
+import {
+  TransactionDirection,
+  TransactionStatus,
+  TransactionType,
+} from '@/entities';
+import { showTransactionDetailsSheet } from '@/handlers/transactions';
+import { getTitle, getTransactionLabel, parseNewTransaction } from '@/parsers';
+import { isZero } from '@/helpers/utilities';
+import { getConfirmedState } from '@/helpers/transactions';
+import { mapNotificationTransactionType } from '@/notifications/mapTransactionsType';
+
+type Callback = () => void;
+
+type Props = PropsWithChildren<{ walletReady: boolean }>;
+
+export const NotificationsHandler = ({ children, walletReady }: Props) => {
+  const wallets = useWallets();
+  const { contacts } = useContacts();
+  const dispatch: ThunkDispatch<AppState, unknown, AnyAction> = useDispatch();
+  const syncedStateRef = useRef({
+    wallets,
+    contacts,
+  });
+  const prevWalletReady = usePrevious(walletReady);
+  const onTokenRefreshListener = useRef<Callback>();
+  const foregroundNotificationListener = useRef<Callback>();
+  const notificationOpenedListener = useRef<Callback>();
+  const appStateListener = useRef<NativeEventSubscription>();
+  const appState = useRef<AppStateStatus>(null);
+  const notifeeForegroundEventListener = useRef<Callback>();
+
+  /*
+  We need to save some properties to a ref in order to have an up-to-date value
+  inside the event listener callbacks closure
+   */
+  syncedStateRef.current.wallets = wallets;
+  syncedStateRef.current.contacts = contacts;
+
+  useEffect(() => {
```

Comment:
Please review this carefully, this is all the logic for subscribing to all notifications

Reply:
i went through methods one by one on dev mode and went over notifee docs to see if there was something obvious not being followed, all looking good 🥇